styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -155,49 +152,12 @@ public String start(WorkflowInstance workflowInstance, RunSpec runSpec) throws I
   }
 
   private void ensureSecrets(WorkflowInstance workflowInstance, RunSpec runSpec) {
-    if (runSpec.serviceAccount().isPresent()) {
-      final String serviceAccount = runSpec.serviceAccount().get();
-      final String secretName = buildSecretName(serviceAccount);
-      final Secret secret = client.secrets().withName(secretName).get();
-      if (secret == null) {
-        final ServiceAccountKey jsonKey;
-        final ServiceAccountKey p12Key;
-        try {
-          jsonKey = serviceAccountKeyManager.createJsonKey(serviceAccount);
-          p12Key = serviceAccountKeyManager.createP12Key(serviceAccount);
-        } catch (IOException e) {
-          logger.error("[AUDIT] Failed to create keys for {}", serviceAccount, e);
-          throw new InvalidExecutionException("Failed to create keys for " + serviceAccount);
-        }
-
-        final Map<String, String> keys = ImmutableMap.of(
-            STYX_WORKFLOW_SA_JSON_KEY, jsonKey.getPrivateKeyData(),
-            STYX_WORKFLOW_SA_P12_KEY, p12Key.getPrivateKeyData()
-        );
-
-        final Map<String, String> annotations = ImmutableMap.of(
-            STYX_WORKFLOW_SA_JSON_KEY_NAME_ANNOTATION, jsonKey.getName(),
-            STYX_WORKFLOW_SA_P12_KEY_NAME_ANNOTATION, p12Key.getName(),
-            STYX_WORKFLOW_SA_SECRET_ANNOTATION, serviceAccount
-        );
-
-        client.secrets().create(new SecretBuilder()
-                                    .withNewMetadata()
-                                    .withName(secretName)
-                                    .withAnnotations(annotations)
-                                    .endMetadata()
-                                    .withData(keys)
-                                    .build());
-        LOG.info("[AUDIT] Secret {} created for {} referred to by workflow {}", secretName,
-                 serviceAccount, workflowInstance.workflowId());
-      } else {
-        LOG.info("[AUDIT] Workflow {} refers to secret {} of {}", workflowInstance.workflowId(),
-                 secretName, serviceAccount);
-      }
-    }
+    ensureServiceAccountKeySecret(workflowInstance, runSpec);
+    ensureCustomSecret(workflowInstance, runSpec);
+  }
 
-    if (runSpec.secret().isPresent()) {
-      final WorkflowConfiguration.Secret specSecret = runSpec.secret().get();
+  private void ensureCustomSecret(WorkflowInstance workflowInstance, RunSpec runSpec) {
+    runSpec.secret().ifPresent(specSecret -> {
       final Secret secret = client.secrets().withName(specSecret.name()).get();
```

Comment:
Feel like to incorporate this check https://github.com/spotify/styx/pull/153/files#diff-d96c43ca98b2c4d50df1a802530ab077R160 ? Or I can open another PR after this one has been merged. Having #153 opened doesn't make much sense due to the massive change brought by this PR.